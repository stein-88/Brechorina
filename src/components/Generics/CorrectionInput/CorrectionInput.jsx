import React from 'react'
import s from './CorrectionInput.scss'

const CorrectionInput = ({ isinValid, mainTarget, customClass }) => (
    <span data-target={mainTarget}>
        <div data-target={mainTarget} className={`${s.correctRight} ${customClass || ''} ${isinValid ? s.inValColor : ''}`}>&nbsp;</div>
        <div data-target={mainTarget} className={`${s.correctLeft} ${customClass || ''} ${isinValid ? s.inValColor : ''}`}>&nbsp;</div>
    </span>
)

export default CorrectionInput
