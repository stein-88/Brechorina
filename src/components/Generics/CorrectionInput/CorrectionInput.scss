.correctRight {
  width: 20px;
  position: absolute;
  height: 25px;
  right: 0;
  top: 31px;
  border: 1px solid #0079d3;
  border-top: none;
  border-bottom: none;
  border-left: none;
  background-color: white;
  z-index: 1;

  &.inValColor {
    border-color: red;
  }
}

.correctLeft {
  width: 20px;
  position: absolute;
  height: 25px;
  top: 31px;
  border: 1px solid #0079d3;
  border-top: none;
  border-bottom: none;
  border-right: none;
  background-color: white;
  z-index: 1;

  &.inValColor {
    border-color: red;
  }
}
