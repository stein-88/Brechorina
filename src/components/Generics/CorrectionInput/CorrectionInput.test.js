import React from 'react'
import renderer from 'react-test-renderer'
import CorrectionInput from './CorrectionInput'

describe('Test on <CorrectionInput /> component', () => {
    test('Should render without props', () => {
        const componentNoprops = renderer.create(<CorrectionInput />)
        const treeNoprops = componentNoprops.toJSON()
        expect(treeNoprops).toMatchSnapshot()
    })
    test('Should render with arabic language props', () => {
        const componentNoprops = renderer.create(<CorrectionInput isinValid mainTarget />)
        const treeNoprops = componentNoprops.toJSON()
        expect(treeNoprops).toMatchSnapshot()
    })
})
