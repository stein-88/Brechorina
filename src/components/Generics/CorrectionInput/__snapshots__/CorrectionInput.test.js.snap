// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <CorrectionInput /> component Should render with arabic language props 1`] = `
<span
  data-target={true}
>
  <div
    className="correctRight  inValColor"
    data-target={true}
  >
     
  </div>
  <div
    className="correctLeft  inValColor"
    data-target={true}
  >
     
  </div>
</span>
`;

exports[`Test on <CorrectionInput /> component Should render without props 1`] = `
<span>
  <div
    className="correctRight  "
  >
     
  </div>
  <div
    className="correctLeft  "
  >
     
  </div>
</span>
`;
